mlflow/genai/judges/utils.py

```diff
@@ -1,40 +1,474 @@
+    assessment_name: str,
+) -> Feedback:
+    """
+    Invoke the Databricks judge using the databricks.agents.evals library.
+
+    Uses the direct chat completions API for clean prompt submission without
+    any additional formatting or template requirements.
+
+    Args:
+        prompt: The formatted prompt with template variables filled in.
+        assessment_name: The name of the assessment.
+
+    Returns:
+        Feedback object from the Databricks judge.
+
+    Raises:
+        MlflowException: If databricks-agents is not installed.
+    """
+    _check_databricks_agents_installed()
+
+    from databricks.rag_eval import context
+
+    try:
+        if isinstance(prompt, str):
+            system_prompt = None
+            user_prompt = prompt
+        else:
+            prompts = _split_messages_for_databricks(prompt)
+            system_prompt = prompts.system_prompt
+            user_prompt = prompts.user_prompt
+
+        @context.eval_context
+        def call_chat_completions():
+            managed_rag_client = context.get_context().build_managed_rag_client()
+
+            llm_result = managed_rag_client.get_chat_completions_result(
+                user_prompt=user_prompt,
+                system_prompt=system_prompt,
+            )
+
+            return _parse_databricks_judge_response(llm_result.output, assessment_name)
+
+        return call_chat_completions()
+
+    except Exception as e:
+        _logger.debug(f"Failed to invoke Databricks judge: {e}")
+        return Feedback(
+            name=assessment_name,
+            error=f"Failed to invoke Databricks judge: {e}",
+            source=AssessmentSource(
+                source_type=AssessmentSourceType.LLM_JUDGE,
+                source_id=_DATABRICKS_DEFAULT_JUDGE_MODEL,
+            ),
+        )
+
+
+def _invoke_via_gateway(
+    model_uri: str,
+    provider: str,
+    messages: list["ChatMessage"],
+) -> str:
+    """
+    Invoke the judge model via native AI Gateway adapters.
+
+    Args:
+        model_uri: The full model URI.
+        provider: The provider name.
+        messages: List of ChatMessage objects.
+
+    Returns:
+        The JSON response string from the model.
+
+    Raises:
+        MlflowException: If the provider is not natively supported or invocation fails.
+    """
+    from mlflow.metrics.genai.model_utils import get_endpoint_type, score_model_on_payload
+
+    if provider not in _NATIVE_PROVIDERS:
+        raise MlflowException(
+            f"LiteLLM is required for using '{provider}' LLM. Please install it with "
+            "`pip install litellm`.",
+            error_code=BAD_REQUEST,
+        )
+
+    return score_model_on_payload(
+        model_uri=model_uri,
+        payload=[{"role": msg.role, "content": msg.content} for msg in messages],
+        endpoint_type=get_endpoint_type(model_uri) or "llm/v1/chat",
+    )
+
+
+@record_usage_event(InvokeCustomJudgeModelEvent)
+def invoke_judge_model(
```

Comment:
This function was the major source of merge conflict when merging master into judges branch. 
@dbczumar please check my work here!!